client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -391,6 +392,21 @@ function mediaButton( editor ) {
 			// edited and marked dirty, we can set the counter to correct number.
 			numOfImagesToUpdate = null;
 		}
+
+		// Re-select image node if one was selected.
```

Comment:
The cause of the problem really isn't clear and this comment doesn't seem to explain it much either. Could we leave more context to indicate why this code exists?